test/videojs-hls_test.js

```diff
@@ -1510,32 +1495,23 @@ test('playlist 404 should end stream with a network error', function() {
   equal(player.tech_.hls.mediaSource.error_, 'network', 'set a network error');
 });
 
-test('segment 404 should trigger MEDIA_ERR_NETWORK', function () {
+test('segment 404 should trigger blacklisting of media', function () {
+  var media;
+
   player.src({
-    src: 'manifest/media.m3u8',
+    src: 'manifest/master.m3u8',
     type: 'application/vnd.apple.mpegurl'
   });
-
   openMediaSource(player);
 
-  standardXHRResponse(requests[0]);
-  requests[1].respond(404);
-  ok(player.tech_.hls.error.message, 'an error message is available');
-  equal(2, player.tech_.hls.error.code, 'Player error code should be set to MediaError.MEDIA_ERR_NETWORK');
-});
-
-test('segment 500 should trigger MEDIA_ERR_ABORTED', function () {
```

Comment:
All non-200 responses are handled by exactly the same code. I didn't see the point of testing each individual code! :D